cpp/tests/test_basic_io.cpp

```diff
@@ -86,3 +97,130 @@ TEST_F(BasicIOTest, write_read_async)
 
   CUDA_DRIVER_TRY(kvikio::cudaAPI::instance().StreamDestroy(stream));
 }
+
+class DirectIOTest : public testing::Test {
+ public:
+  using value_type = std::int64_t;
+
+ protected:
+  void SetUp() override
+  {
+    TempDir tmp_dir{false};
+    _filepath = tmp_dir.path() / "test";
+
+    // Skip the fixture is Direct I/O is not supported
+    try {
+      [[maybe_unused]] auto fd =
+        kvikio::open_fd(_filepath.c_str(), "w", true /* o_direct */, kvikio::FileHandle::m644);
+    } catch (...) {
+      GTEST_SKIP() << "Direct I/O is not supported for the test file: " << _filepath;
+    }
+
+    // Create a sequence of numbers as a ground truth
+    _num_elements = 8ULL * 1024ULL * 1024ULL + 1234ULL;
```

Comment:
this is almost 40GB of data, right?
How long does it take to run these tests (including the fixture init)?